cmd/whisperImporter/chunkWriter/main.go

```diff
@@ -0,0 +1,165 @@
+package main
+
+import (
+	"compress/gzip"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/gocql/gocql"
+	"github.com/raintank/dur"
+	"github.com/raintank/metrictank/cmd/whisperImporter"
+	"github.com/raintank/metrictank/idx"
+	"github.com/raintank/metrictank/idx/cassandra"
+	"github.com/raintank/metrictank/mdata"
+	"gopkg.in/raintank/schema.v1"
+)
+
+const table_name_format = `metric_%d`
+
+var (
+	httpEndpoint      = flag.String("http-endpoint", "127.0.0.1:8080", "The http endpoint to listen on")
+	threads           = flag.Int("threads", 10, "Number of workers threads to process .wsp files")
+	exitOnError       = flag.Bool("exit-on-error", true, "Exit with a message when there's an error")
+	cassandraAddrs    = flag.String("cassandra-addrs", "localhost", "cassandra host (may be given multiple times as comma-separated list)")
+	cassandraKeyspace = flag.String("cassandra-keyspace", "metrictank", "cassandra keyspace to use for storing the metric data table")
+	cassAddr          = flag.String("cass-addr", "localhost", "Address of cassandra host.")
+	keyspace          = flag.String("keyspace", "raintank", "Cassandra keyspace to use.")
+	ttlsStr           = flag.String("ttls", "35d", "list of ttl strings used by MT separated by :")
+	windowFactor      = flag.Int("window-factor", 20, "the window factor be used when creating the metric table schema")
+	partitionBy       = flag.String("partition-by", "bySeries", "Method used for paritioning metrics. (byOrg|bySeries)")
+	numPartitions     = flag.Int("num-partitions", 0, "Number of Partitions")
+)
+
+func throwError(msg string) {
+	panic(msg)
+}
+
+type Server struct {
+	Cluster     *gocql.ClusterConfig
+	TTLTables   mdata.TTLTables
+	Partitioner sarama.Partitioner
+	Index       idx.MetricIndex
+}
+
+func (s *Server) getPartition(md *schema.MetricData) int32 {
+	var key []byte
+	switch *partitionBy {
+	case "byOrg":
+		key = md.KeyByOrgId(nil)
+	case "bySeries":
+		key = md.KeyBySeries(nil)
+	}
+	partition, _ := s.Partitioner.Partition(&sarama.ProducerMessage{Key: sarama.ByteEncoder(key)}, int32(*numPartitions))
+	return partition
+}
+
+func decode(body io.Reader) *whisperImporter.Metric {
+	gzipReader, err := gzip.NewReader(body)
+	if err != nil {
+		throwError(fmt.Sprintf("Error creating gzip reader: %s", err))
+	}
+
+	raw, err := ioutil.ReadAll(gzipReader)
+	if err != nil {
+		throwError(fmt.Sprintf("Error reading from gzip reader: %s", err))
+	}
+
+	metric := whisperImporter.Metric{}
+
+	err = json.Unmarshal(raw, &metric)
+	if err != nil {
+		throwError(fmt.Sprintf("Error unmarshaling json: %s", err))
+	}
+
+	return &metric
+}
+
+func (s *Server) chunksHandler(w http.ResponseWriter, req *http.Request) {
+	metric := decode(req.Body)
+
+	session, err := s.Cluster.CreateSession()
```

Comment:
You're right. In fact the docs say (to my surprise) that a session is thread safe and can be used by many routines, so I'll instantiate one and put it in the `Server struct`.